bindata/bootkube/bootstrap-manifests/etcd-member-pod.yaml

```diff
@@ -43,41 +43,38 @@ spec:
 
       source /run/etcd/environment
 
-      # TODO remove when this is handeled in setup-etcd-env
-      ETCD_IP_ADDRESS=${ETCD_IPV4_ADDRESS:-$ETCD_IPV6_ADDRESS}
-
-      [ -e /etc/ssl/etcd/system:etcd-server:${ETCD_IP_ADDRESS}.crt -a \
-        -e /etc/ssl/etcd/system:etcd-server:${ETCD_IP_ADDRESS}.key ] || \
+      [ -e /etc/ssl/etcd/system:etcd-server:${ETCD_ESCAPED_IP_ADDRESS}.crt -a \
+        -e /etc/ssl/etcd/system:etcd-server:${ETCD_ESCAPED_IP_ADDRESS}.key ] || \
         kube-client-agent \
           request \
             --kubeconfig=/etc/kubernetes/kubeconfig \
             --orgname=system:etcd-servers \
             --assetsdir=/etc/ssl/etcd \
             --dnsnames={{.EtcdServerCertDNSNames}} \
-            --commonname=system:etcd-server:${ETCD_IP_ADDRESS} \
+            --commonname=system:etcd-server:${ETCD_ESCAPED_IP_ADDRESS} \
```

Comment:
is this the common name of the certificate that was signed above?  If so, I think you should retrieve it from the cert, not count on the cert to be matched in a particular way. This constrains future migration choice.

Reply:
> is this the common name of the certificate that was signed above? If so, I think you should retrieve it from the cert, not count on the cert to be matched in a particular way. This constrains future migration choice.

The agent is requesting is sending a CSR based on the commonname defined. So this action against the`request` command is before certs exist.